ignite/engine/__init__.py

```diff
@@ -335,11 +345,11 @@ def create_supervised_trainer(
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
-    mode = _check_arg(on_tpu, amp_mode, scaler)
+    mode, _scaler = _check_arg(on_tpu, amp_mode, scaler)
 
     if mode == "amp":
         _update, _scaler = supervised_training_step_amp(
```

Comment:
```suggestion
        _update = supervised_training_step_amp(
```

No need to return the scaler now, right ?